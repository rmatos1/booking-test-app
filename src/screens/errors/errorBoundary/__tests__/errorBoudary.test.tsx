import { describe, test } from '@jest/globals';
import { render } from "@testing-library/react";

import { ErrorBoundary, IErrorBoundaryProps } from "..";



const defaultProps: IErrorBoundaryProps = {
    children: <div data-testid="children-element" />
}

const ErrorThrower = () => {
    throw new Error('Test error');
  };

const setup = (componentProps?: IErrorBoundaryProps): JSX.Element => {

    const baseProps = componentProps || defaultProps;

  return(
    <ErrorBoundary {...baseProps} />
  )
};

describe('<ErrorBoundary />', () => {
  test('should render correctly the defined children element', () => {
    const wrapper = render(setup());

    const children = wrapper.getByTestId("children-element");

    expect(children).toBeDefined()
    expect(wrapper).toMatchSnapshot();
  });

  test("should render the error page whenever an error happens",  () => {
   
    const wrapper = render(setup({ children: <div><ErrorThrower  /></div> }));

    const gotErrorTitle = wrapper.getByTestId("got-error-title")

    expect(gotErrorTitle).toBeDefined()
    expect(wrapper).toMatchSnapshot();
  })
});
