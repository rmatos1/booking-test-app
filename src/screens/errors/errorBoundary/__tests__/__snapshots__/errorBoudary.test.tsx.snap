// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBoundary /> should render correctly the defined children element 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="children-element"
      />
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="children-element"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<ErrorBoundary /> should render the error page whenever an error happens 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bqyKbq gCCXA-D"
      >
        <img
          alt="Error app"
          class="sc-eCstZk cgIxof"
          data-testid="image"
        />
        <div
          class="sc-kfzCjt KTkhM"
        >
          <h1
            class="sc-fKFygU dbnRdB"
            data-testid="got-error-title"
          >
            We got an error!
          </h1>
          <p>
            Sorry, we got a momentary malfunctioning.
            

            Click on the link below to refresh the page
          </p>
          <a
            class="sc-cxFKTC jNYELN"
            href="/"
          >
            New Booking
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bqyKbq gCCXA-D"
    >
      <img
        alt="Error app"
        class="sc-eCstZk cgIxof"
        data-testid="image"
      />
      <div
        class="sc-kfzCjt KTkhM"
      >
        <h1
          class="sc-fKFygU dbnRdB"
          data-testid="got-error-title"
        >
          We got an error!
        </h1>
        <p>
          Sorry, we got a momentary malfunctioning.
          

          Click on the link below to refresh the page
        </p>
        <a
          class="sc-cxFKTC jNYELN"
          href="/"
        >
          New Booking
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
